modules/spring-config/src/main/java/org/mule/runtime/config/internal/factories/ComponentAstSpliteratorStrategyFactory.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.config.internal.factories;
+
+import static org.mule.runtime.ast.api.util.AstTraversalDirection.BOTTOM_UP;
+import static org.mule.runtime.ast.api.util.AstTraversalDirection.TOP_DOWN;
+
+import org.mule.runtime.ast.api.ComponentAst;
+import org.mule.runtime.ast.api.util.AstTraversalDirection;
+import org.mule.runtime.config.internal.model.iterator.BottomUpComponentAstSpliteratorTraversal;
+import org.mule.runtime.config.internal.model.iterator.TopDownComponentAstSpliteratorTraversal;
+
+import java.util.Spliterator;
+
+/**
+ * Factory that knows how to create a {@link Spliterator<ComponentAst>} to navigates the whole AST.
+ *
+ * @since 4.4.0
+ */
+public class ComponentAstSpliteratorStrategyFactory {
```

Comment:
This should be `AstTraversalDirection` directly instead of having a separate factory.